File: API/Models/LectionModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace API.Models
{
    public class LectionModel
    {
        [Required]
        public int LectionID { get; set; }
        public string LectionType { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string LectionBody { get; set; }
        public int MaxMark { get; set; }
        public bool AdditionalMarkIsAvailable { get; set; }
        public TeacherModel Creator { get; set; }
        public DateTime CreationDate { get; set; }
        public List<CourseModel> Courses { get; set; } = new List<CourseModel>();
        public DateTime? Date { get; set; }
        public LectionModel()
        {

        }
    }
}

File: EduAngular/src/app/shared/lection.model.ts
import { Teacher } from './teacher.model';
import { Course } from './course.model';

export class Lection {
    LectionID : number;
    LectionType : string;
    Name : string;
    Description : string;
    LectionBody : string;
    MaxMark : number;
    AdditionalMarkIsAvailable : boolean;
    Creator : Teacher;
    CreationDate : Date;
    Courses : Course;
    Date : Date;
}
